spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractFetchLimitingMessageSource.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.endpoint;
+
+/**
+ * A message source that can limit the number of remote objects it fetches.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public abstract class AbstractFetchLimitingMessageSource<T> extends AbstractMessageSource<T> {
+
+	private volatile int maxFetchSize = Integer.MIN_VALUE;
+
+	/**
+	 * Set the maximum number of objects the source should fetch if it is necessary
+	 * to fetch objects. The source must be a {@link LimitingMessageSource}.
```

Comment:
There is no such an interface any more. Therefore this JavaDoc must be fixed.
But I guess we should introduce `FetchLimitingMessageSource` anyway.
Remember we can proxy the entire `MessageSource` with the `AbstractMessageSourceAdvice`?
Having that we won't be able to change `setMaxFetchSize` at runtime, when it is a proxy.